### ISSUE TITLE
web.py - SAWarning: The IN-predicate on "sightings.pokemon_id" was invoked with an empty sequence.

### ISSUE THREAD

**Issue by pokemonracer** [modrzew/pokeminer#93]
Got this error in my web.py-console. Could this be problem with my sqlite-db being too slow for my workers?

> Python27\lib\site-packages\sqlalchemy\sql\default_comparator.py:153: SAWarning: The IN-predicate on "sightings.pokemon_id" was invoked with an empty sequence. This results in a contradiction, which nonetheless can be expensive to evaluate. Consider alternative strategies for improved performance. 'strategies for improved performance.' % expr)

Also logging for web.py would be awesome! :)

**Comment by Aiyubi** [comment]
I am getting the same error on one of my machines (with mysql) - only one of them gets this error while the others run fine.

Also logging for web? Like the stuff you see on your console?

**Comment by pokemonracer** [comment]
Haven't seen this error since. Odd.

And yes, i mean stuff on the console, just like with worker.py.

**Comment by Aiyubi** [comment]
as far as I understand it, this was one of the exception cases that got already fixed without looking at this bug

**Comment by GaelicGamer** [comment]
This should already be fixed, try pulling the latest master and see if it still occurs, if it does it sounds like the worker found no pokemon and the duplicate check is throwing the error due to an empty set in its where clause that includes the list of pokemon_ids.  See lines 123-130 of db.py:

```
    existing = session.query(Sighting) \
        .filter(Sighting.pokemon_id == obj.pokemon_id) \
        .filter(Sighting.spawn_id == obj.spawn_id) \
        .filter(Sighting.expire_timestamp > obj.expire_timestamp - 10) \
        .filter(Sighting.expire_timestamp < obj.expire_timestamp + 10) \
        .filter(Sighting.lat == obj.lat) \
        .filter(Sighting.lon == obj.lon) \
        .first()
```

**Comment by modrzew** [comment]
That error is most likely because your STAGE2 list is empty. It should be gone after you fill it up.

**Comment by modrzew** [comment]
Aaaaand it should be fixed as of 55e30dd706cb64c8fd53abca7da8f2df7fbaa7fc.
